src/main/java/Deadline.java

```diff
@@ -0,0 +1,17 @@
+/**
+ * Represents a task that has a deadline.
+ * @author Aaron
+ */
+public class Deadline extends Task {
+    protected String by;
```

Comment:
you can consider changing 'by' to something like doByDate so that it is clearer for the reader.